x-pack/plugins/cases/public/components/user_actions/use_user_actions_handler.tsx

```diff
@@ -70,14 +71,33 @@ export const useUserActionsHandler = (): UseUserActionsHandler => {
     [clearDraftComment]
   );
 
+  const removeCommentIdFromLoadingIds = (commentId: string) => {
+    setLoadingCommentIds((ids) => ids.filter((loadingId) => loadingId !== commentId));
+  };
+
+  const addCommentIdToLoadingIds = (commentId: string) => {
+    setLoadingCommentIds((ids) => [...ids, commentId]);
+  };
+
   const handleSaveComment = useCallback(
     ({ id, version }: { id: string; version: string }, content: string) => {
-      patchComment({
-        caseId,
-        commentId: id,
-        commentUpdate: content,
-        version,
-      });
+      addCommentIdToLoadingIds(id);
+      patchComment(
+        {
+          caseId,
+          commentId: id,
+          commentUpdate: content,
+          version,
+        },
+        {
```

Comment:
since we moved handling loadingCommentIds logic out of `use_update_comment` hook,  Will we have to handle loadingCommentIds everywhere we use patchComment?
I am just wondering what will be the impact if someone uses patchComment somewhere else and doesn't add or remove loadingCommentIds.

Reply:
yeah, it makes sense. Currently it's action handler's responsibility to handle comments. And if it changes in future, we can also take care of `loadingCommentIds` accordingly.